opts/hostcfg.go

```diff
@@ -220,6 +229,13 @@ func (h *HostCfg) UnmarshalJSON(data []byte) error {
 		h.ProvisioningURLs = urls
 	}
 
+	if tricky.Timestamp != 0 {
+		if h.Timestamp == nil {
+			h.Timestamp = new(time.Time)
+		}
+		*h.Timestamp = time.Unix(tricky.Timestamp, 0)
+	}
```

Comment:
Can't you assign `&time.Unix(tricky.Timestamp, 0)` to `h.Timestamp` even if it is `nil`?!
Just 
```
if tricky.Timestamp != 0 {
	u := time.Unix(tricky.Timestamp, 0)
	h.Timestamp = &u
}